x/sync/client.go

```diff
@@ -268,8 +265,8 @@ func (c *client) GetRangeProof(
 			ctx,
 			&rangeProof,
 			int(req.KeyLimit),
-			startKey,
-			endKey,
+			maybeBytesToMaybe(req.StartKey),
+			maybeBytesToMaybe(req.EndKey),
```

Comment:
I think we could pull this into a separate small cleanup PR

Reply:
https://github.com/ava-labs/avalanchego/pull/2269